libs/langchain/langchain/chat_models/base.py

```diff
@@ -10,4 +14,179 @@
     "SimpleChatModel",
     "generate_from_stream",
     "agenerate_from_stream",
+    "init_model",
 ]
+
+
+@beta()
+def init_model(
+    model_name: str, *, model_provider: Optional[str] = None, **kwargs: Any
+) -> BaseChatModel:
+    """Initialize a ChatModel from the model name and provider.
+
+    Must have the integration package corresponding to the model provider installed.
+
+    Args:
+        model_name: The name of the model, e.g. "gpt-4o", "claude-3-opus-20240229".
+        model_provider: The model provider. Will attempt to infer from model_name if
+            not specified. Valid provider values are:
+            - openai
+            - anthropic
+            - azure_openai
+            - cohere
+            - google_vertexai
+            - google_genai
+            - fireworks
+            - ollama
+            - together
+            - mistralai
+            - huggingface
+            - groq
+            - bedrock
+        **kwargs: Additional keyword args to pass to
+            <<selected ChatModel>>.__init__(model=model_name, **kwargs).
+
+    Returns:
+        The BaseChatModel corresponding to the model_name and model_provider specified.
+
+    Raises:
+        ValueError: If model_provider cannot be inferred or isn't supported.
+        ImportError: If the model provider integration package is not installed.
+
+    Example:
+        .. code-block:: python
+
+            from langchain.chat_models import init_model
+
+            gpt_3 = init_model("gpt-4o", model_provider="openai", temperature=0)
+            claude_opus = init_model("claude-3-opus-20240229", model_provider="anthropic", temperature=0)
+            gemini_15 = init_model("gemini-1.5-pro", model_provider="google_vertexai", temperature=0)
+
+            gpt_3.invoke("what's your name")
+            claude_opus.invoke("what's your name")
+            gemini_15.invoke("what's your name")
+    """  # noqa: E501
+    model_provider = model_provider or _attempt_infer_model_provider(model_name)
```

Comment:
What qualifications must a chat model meet to be included here?

Reply:
recognized model_provider below

Reply:
any ChatModel can be added. can update with more if we want

Reply:
until requested will stick only to models with their own integration packages, not langchain-community ones